src/components/MyAccount/RequestsTab/UpdateLocation.tsx

```diff
@@ -0,0 +1,196 @@
+// Modal onClose
+import {
+  Box,
+  Icon,
+  useModal,
+  Text,
+  Heading,
+  Button,
+  Select,
+  Link as DSLink,
+  type DefaultModalProps,
+  type ConfirmationModalProps,
+  type BaseModalProps,
+  SkeletonLoader,
+} from "@nypl/design-system-react-components"
+import type { SierraCodeName } from "../../../types/myAccountTypes"
+import styles from "../../../../styles/components/MyAccount.module.scss"
+import { useState } from "react"
+import { BASE_URL } from "../../../config/constants"
+
+interface UpdateLocationPropsType {
+  holdId: string
+  pickupLocation: SierraCodeName
+  pickupLocationOptions: SierraCodeName[]
+  key: number
+  patronId: number
+  updateHoldLocation: (holdId: string, location: SierraCodeName) => void
+}
+
+const UpdateLocation = ({
+  updateHoldLocation,
+  pickupLocationOptions,
+  patronId,
+  holdId,
+  pickupLocation,
+  key,
+}: UpdateLocationPropsType) => {
+  const { Modal, onOpen: openModal, onClose: closeModal } = useModal()
+  const locationsWithSelectedFirst = [
+    pickupLocation,
+    ...pickupLocationOptions.filter(
+      (loc) => loc.code.trim() !== pickupLocation.code.trim()
+    ),
+  ]
+  const confirmLocationChangeModalProps = (selected: SierraCodeName) => ({
+    type: "confirmation",
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Select
+          value={selected.code}
+          onChange={(e: React.FormEvent<HTMLInputElement>) => {
+            const newLocation = locationsWithSelectedFirst.find(
+              (loc) => e.currentTarget.value === loc.code
+            )
+            setModalProps(
+              confirmLocationChangeModalProps(
+                newLocation
+              ) as ConfirmationModalProps
+            )
+          }}
+          id={`update-location-selector-${key}`}
+          labelText="Pickup location"
+          showLabel
+          name={"pickup-location"}
+        >
+          {locationsWithSelectedFirst.map((location, locationKey) => {
+            return (
+              <option
+                key={`pickupLocation-option-${key}${locationKey}`}
+                value={location.code}
+              >
+                {location.name}
+              </option>
+            )
+          })}
+        </Select>
+      </Box>
+    ),
+    closeButtonLabel: "Cancel location change",
+    confirmButtonLabel: "Confirm location",
+    onCancel: () => {
+      closeModal()
+    },
+    onConfirm: async () => {
+      setModalProps({
+        ...confirmLocationChangeModalProps(selected),
+        bodyContent: <SkeletonLoader showImage={false} />,
```

Comment:
what do you think about the heading also being set to have the skeleton loader?

Reply:
I think the heading is necessary to populate some aria attributes, so I don't want to remove it here.